packages/cli/src/Server.ts

```diff
@@ -711,6 +754,64 @@ class App {
 			};
 		}));
 
+		// Retrieves all tags, with or without usage count
+		this.app.get(`/${this.restEndpoint}/tags`, ResponseHelper.send(async (req: express.Request, res: express.Response): Promise<ITagDb[] | Array<{ id: string, name: string, usageCount?: number }>> => {
```

Comment:
just curious, how does authentication work here? is there a middleware somewhere that handles authentication for all rest endpoints? can you point me to that?

Reply:
Looks like it basic auth is implemented [here](https://github.com/n8n-io/n8n/blob/d59e6d1c4b41142495d5ca1ed0bdfdfcf6d66286/packages/cli/src/Server.ts#L224-L275) and JWT auth is implemented [here](https://github.com/n8n-io/n8n/blob/d59e6d1c4b41142495d5ca1ed0bdfdfcf6d66286/packages/cli/src/Server.ts#L277-L308). Docs reference [here](https://docs.n8n.io/reference/security.html).

Reply:
thanks for the info

Reply:
just wanted to confirm that authentication applies by default to these new endpoints. seems to be the case. 

only skipped for these endpoints https://github.com/n8n-io/n8n/blob/d59e6d1c4b41142495d5ca1ed0bdfdfcf6d66286/packages/cli/src/Server.ts#L219